creator-node/src/services/PrometheusMetric.js

```diff
@@ -0,0 +1,143 @@
+const client = require('prom-client')
+
+const PrometheusType = Object.freeze({
+  HISTOGRAM: 'histogram',
+  GAUGE: 'gauge'
+})
+
+/**
+ * Class to maintain library-specific logic
+ * for Prometheus-bound metrics
+ */
+class PrometheusMetric {
+  // use these static vars to avoid name collisions
+  // and register collectors to be run by the PrometheusMetricExporter
+  static histograms = {}
+  static gauges = {}
+  static registeredCollectors = {}
+
+  /**
+   * Helper to avoid naming collisions.
+   */
+  _initMetric(
+    name,
+    description,
+    labelNames,
+    collection,
+    PrometheusMetricClass
+  ) {
+    if (!(name in collection)) {
+      collection[name] = new PrometheusMetricClass({
+        name: name,
+        help: description,
+        labelNames: labelNames
+      })
+    }
+    this.metric = collection[name]
+  }
+
+  constructor(name, description, labelNames, metric_type) {
+    // set this.startTime to now()
+    this.resetTimer()
+
+    // set metric prefix of audius_project_
+    name = `audius_cn_${name}`
+
+    this.metricType = metric_type
+    if (this.metricType === undefined) {
+      this.metricType = PrometheusType.HISTOGRAM
+    }
+
+    // CollectorRegistries must be uniquely named
```

Comment:
whats a CollectorRegistry?

Reply:
We're not allowed to call `new PrometheusMetricClass({name})` twice with the same name.

Instead, we will use something like a Python `defaultdict` to create and return `new PrometheusMetricClass({name})` if we haven't seen it before, or the previously created instance if we have seen it before.

A good hierarchical overview is:

* Prometheus Registries hold Collectors.
* Collectors hold registers of the metrics that they are holding.
* Exporters export all metrics held within registries for each collector held within the main registry.